src/containerapp/azext_containerapp/custom.py

```diff
@@ -2469,21 +2479,33 @@ def containerapp_up(cmd,
                     context_path=None,
                     service_principal_client_id=None,
                     service_principal_client_secret=None,
-                    service_principal_tenant_id=None):
+                    service_principal_tenant_id=None,
+                    custom_location=None,
+                    connected_cluster_id=None):
     from ._up_utils import (_validate_up_args, _reformat_image, _get_dockerfile_content, _get_ingress_and_target_port,
-                            ResourceGroup, ContainerAppEnvironment, ContainerApp, _get_registry_from_app,
+                            ResourceGroup, ContainerAppEnvironment, ContainerApp, CustomLocation, _get_registry_from_app,
                             _get_registry_details, _create_github_action, _set_up_defaults, up_output,
-                            check_env_name_on_rg, get_token, _validate_containerapp_name)
+                            check_env_name_on_rg, get_token, _validate_containerapp_name, format_location)
     from ._github_oauth import cache_github_token
     HELLOWORLD = "mcr.microsoft.com/azuredocs/containerapps-helloworld"
     dockerfile = "Dockerfile"  # for now the dockerfile name must be "Dockerfile" (until GH actions API is updated)
-
+    location = format_location(location)
     _validate_containerapp_name(name)
 
     register_provider_if_needed(cmd, CONTAINER_APPS_RP)
     _validate_up_args(cmd, source, image, repo, registry_server)
     validate_container_app_name(name)
-    check_env_name_on_rg(cmd, env, resource_group_name, location)
+    if custom_location or connected_cluster_id:
+        register_provider_if_needed(cmd, CUSTOM_LOCATION_RP)
+        register_provider_if_needed(cmd, KUBERNETES_CONFIGURATION_RP)
```

Comment:
Move the validation logic into the _validate_up_args method

Reply:
done